appservice/src/intTest/groovy/nu/yona/server/AbstractAppServiceIntegrationTest.groovy

```diff
@@ -137,28 +138,50 @@ abstract class AbstractAppServiceIntegrationTest extends Specification
 		assert response.status == 200
 	}
 
-	void addTimeZoneGoal(User user, activityCategoryURL, zones, relativeCreationDateTimeString)
+	TimeZoneGoal addTimeZoneGoal(User user, activityCategoryURL, zones, relativeCreationDateTimeString)
 	{
 		ZonedDateTime creationTime = YonaServer.relativeDateTimeStringToZonedDateTime(relativeCreationDateTimeString)
 		addTimeZoneGoal(user, activityCategoryURL, zones, creationTime)
 	}
 
-	void addTimeZoneGoal(User user, activityCategoryURL, zones, ZonedDateTime creationTime = YonaServer.now)
+	TimeZoneGoal addTimeZoneGoal(User user, activityCategoryURL, zones, ZonedDateTime creationTime = YonaServer.now)
 	{
 		appService.addGoal(appService.&assertResponseStatusCreated, user, TimeZoneGoal.createInstance(creationTime, activityCategoryURL, zones.toArray()))
 	}
 
-	void addBudgetGoal(User user, activityCategoryURL, int maxDurationMinutes, relativeCreationDateTimeString)
+	void updateTimeZoneGoal(User user, TimeZoneGoal originalGoal, zones, relativeCreationDateTimeString)
```

Comment:
You could change this void in `TimeZoneGoal`. Same for the four following methods.